lucene/core/src/java/org/apache/lucene/util/QueryBuilder.java

```diff
@@ -450,9 +485,13 @@ protected Query analyzePhrase(String field, TokenStream stream, int slop) throws
         position += 1;
       }
       builder.add(new Term(field, termAtt.getBytesRef()), position);
+      phraseBoost = boostAtt.getBoost();
```

Comment:
I think this isn't quite right, because we need to combine boosts together somehow; currently your phrase boost is just the boost of the last term in the phrase.

Reply:
I implemented a simple multiplicative boost.
It's back compatible with the designed use case (multi term synonym -> single concept -> single boost -> e.g. panthera onca => jaguar|0.95, big cat|0.85, black panther|0.65))

But it's also compatible in not synonym cases, if the user needs a boost per token in phrase and span queries.
It's in the upcoming commit, let me know if you believe something different is necessary